### PR TITLE
Automatically delete Transifex resources

### PR DESCRIPTION
**Pull request by camertron**
Currently, resources in Transifex are never deleted, which can lead to a build-up of crusty, unused resources. This PR adds a handler for the Github `delete` event, which gets fired whenever a branch or tag is deleted. Since it's common to delete a branch after merging to master, this functionality should prevent the accumulation of old resources.

@lumoslabs/platform

### CODE REVIEW

File: lib/txgh/github_repo.rb [hunk 1]
```diff
@@ -19,6 +19,22 @@ def process_all_branches?
       branch == 'all'
     end
 
+    def should_process_branch?(candidate)
+      process_all_branches? ||
+        candidate.include?(github_config_branch) ||
+        candidate.include?('L10N')
```
Comment (11mdlow) [inline lib/txgh/github_repo.rb:25]: might be good to document what's so special about this particular branch name/assign it to a constant

File: lib/txgh/handlers/github/delete_handler.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
+require 'logger'
+
+module Txgh
+  module Handlers
+    module Github
+      class DeleteHandler
+        include ResponseHelpers
+
+        attr_reader :project, :repo, :payload, :logger
```
Comment (11mdlow) [inline lib/txgh/handlers/github/delete_handler.rb]: seems like each of the handlers have these attributes in common, might be good to create a parent `Handler` class and then subclass it with each handler. You don't necessarily have to do it in this PR, but might be a good refactor.

File: lib/txgh/github_repo.rb (hunk 1 above)
Comment (camertron) [inline lib/txgh/github_repo.rb:25]: Yeah... I'm actually not sure what this is for. We inherited it from Transifex. Maybe I should just get rid of it in our fork?

File: lib/txgh/handlers/github/delete_handler.rb (hunk 2 above)
Comment (camertron) [inline lib/txgh/handlers/github/delete_handler.rb]: Yeah, I think that makes sense.

**Comment by 11mdlow** [comment]
sweet, I like it :+1:

**Pull request merged**